### PR TITLE
Make sure Addon.compatible_version only returns listed versions

### PR DESCRIPTION
**Pull request by diox**
The legacy API (called by the add-on manager in Firefox) uses that method.

### CODE REVIEW

File: src/olympia/addons/models.py [hunk 1]
```diff
@@ -825,8 +825,13 @@ def compatible_version(self, app_id, app_version=None, platform=None,
...
                                        compat_mode))
         valid_file_statuses = ','.join(map(str, self.valid_file_statuses))
-        data = dict(id=self.id, app_id=app_id, platform=platform,
-                    valid_file_statuses=valid_file_statuses)
+        data = {
+            'id': self.id,
+            'app_id': app_id,
+            'platform': platform,
+            'valid_file_statuses': valid_file_statuses,
+            'channel': amo.RELEASE_CHANNEL_LISTED,
```
Comment (eviljeff) [inline src/olympia/addons/models.py]: To confirm, this method is never used by unlisted versions, e.g. in devhub.

File: src/olympia/addons/tests/test_models.py [hunk 2]
```diff
@@ -541,23 +541,16 @@ def test_current_beta_version(self):
         addon = Addon.objects.get(pk=5299)
         assert addon.current_beta_version.id == 50000
 
-    def _create_new_version(self, addon, status):
```
Comment (eviljeff) [inline src/olympia/addons/tests/test_models.py]: 💯

File: src/olympia/addons/models.py (hunk 1 above)
Comment (diox) [inline src/olympia/addons/models.py]: Thankfully no. I considered moving it to `legacy_api/utils.py` or something in fact. Think that would be better, for our sanity ?

Comment (eviljeff) [inline src/olympia/addons/models.py]: Anything legacy we can move out of the Addon model would be good for comprehensibility.  Up to you if you want to do it in this commit though.

Comment (diox) [inline src/olympia/addons/models.py]: ff37b5801d5a6423828802bc28dbdd9bef3952b9

**Pull request merged**